test/lib/token.test.js

```diff
@@ -0,0 +1,117 @@
+'use strict';
+
+const assert = require('chai').assert;
+const sinon = require('sinon');
+const mockery = require('mockery');
+const schema = require('screwdriver-data-schema');
+
+sinon.assert.expose(assert, { prefix: '' });
+
+describe('Token Model', () => {
+    const password = 'password';
+    let BaseModel;
+    let TokenModel;
+    let ironMock;
+    let datastore;
+    let createConfig;
+    let token;
+
+    before(() => {
+        mockery.enable({
+            useCleanCache: true,
+            warnOnUnregistered: false
+        });
+    });
+
+    beforeEach(() => {
+        datastore = {
+            update: sinon.stub()
+        };
+        datastore.update.resolves({});
+
+        ironMock = {
+            seal: sinon.stub(),
+            unseal: sinon.stub(),
+            defaults: {}
+        };
+        mockery.registerMock('iron', ironMock);
+
+        // eslint-disable-next-line no-global-assign
+        Date = function Date() {
```

Comment:
You wouldn't think it would be a big deal until you spend two days trying to figure out why `async.waterfall` starts causing tests to timeout because it is waiting on a `process.nextTick` that never comes.